src/mono/wasm/runtime/invoke-js.ts

```diff
@@ -29,6 +29,70 @@ export function mono_wasm_bind_js_function(function_name: MonoStringRef, module_
         const fn = mono_wasm_lookup_function(js_function_name, js_module_name);
         const args_count = get_signature_argument_count(signature);
 
+        const arg_marshalers: Function[] = [];
+        const arg_cleanup: Function[] = [];
+        for (let index = 0; index < args_count; index++) {
+            const sig = get_sig(signature, index + 2);
+            const marshaler_type = get_signature_type(sig);
+            const arg_marshaler = bind_arg_marshal_to_js(sig, marshaler_type, index + 2);
+            arg_marshalers.push(arg_marshaler);
+            arg_cleanup.push(marshaler_type !== MarshalerType.Span
```

Comment:
one good option would be to only push an arg_cleanup function if cleanup needs to happen, then the array doesn't contain any dummy functions or nulls and the loop in bound_fn can run however many are there.